dbms/src/Storages/Page/V3/CheckpointFile/CPFilesWriter.cpp

```diff
@@ -0,0 +1,130 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Storages/Page/V3/CheckpointFile/CPFilesWriter.h>
+
+namespace DB::PS::V3
+{
+
+CPFilesWriter::CPFilesWriter(CPFilesWriter::Options options)
+    : manifest_file_id(options.manifest_file_id)
+    , data_writer(CPDataFileWriter::create({
+          .file_path = options.data_file_path,
+          .file_id = options.data_file_id,
+      }))
```

Comment:
`CPDataFileWriter::Options::file_id` should be the lockkey to the CheckpointDataFile, which is generated by `S3FilenameView:: getLockKey`.

What about renaming it to `CPDataFileWriter::Options::lock_file`?

https://github.com/pingcap/tiflash/blob/da5c10dd76cc76ceba4cfad2c538df0de40d7001/dbms/src/Storages/S3/S3Filename.h#L88-L89

Reply:
I'm not sure. Seems that the relationship is that, the lock key should be identical to the data file id, instead of in the reserve way.

Data file id should be directly tied with Data file path, as a replacement (or shorten, or a format easy to deal with) for the Data file path.